### DISCUSSION TITLE
Expanding the ltsp target group

### DISCUSSION THREAD

**Discussion by rkwesk** [ltsp/ltsp#182] [Q&A]
I have always considered linux as one of the best kept secrets and within those who do know about linux the ltsp model is also a best kept secret.

Of course ltsp is not appropriate everywhere. But I think that once those who would benefit were shown they would embrace ltsp.

In my limited experience as an IT teacher for 8 years until I retired in 2001 I experience firsthand the tedious work of keeping the workstations in sync and dealing with viruses. Since then I have helped some IT teachers in Greece with ltsp implementation and have received a great deal of positive feedback.

The issue I am raising is that this project needs showing and explaining to people. At the moment most of the ltsp users are using but not sharing. Also, gui frontends are far easier for many and I consider the ltsp manager that was first put out for ltsp5 as a good example.

Yes, among those of us here and most others at the ltsp5 list, few are interested in such frontends. I myself have always found the commandline faster and more informative but then I started with it years ago and I want and use the feedback the commandline provides. However, if we consider those who are potentially ltsp users and others whose time is so limited that the commandline seems counter productive to them, then imho such frontends are called for.

Richard

**Comment by alkisg** [comment]
LTSP Manager, which was an internationalized version of sch-scripts, had two roles:
(1) Powerful user manager: ability to create e.g. 50 users x 10 classrooms with a couple of clicks; or select users with the mouse and delete them, etc etc.
(2) LTSP manager

In the ancient past, it also had:
(3) Classroom (client) manager

(3) Was separated into the "epoptes" application.
It would be very useful if (1) was also extracted as an application, completely unrelated to LTSP.

So the LTSP specific part is just (2). This mainly offers the following:
(a) A menu to run commands like `ltsp image`, `ltsp ipxe`, `ltsp dnsmasq`, and
(b) A menu to edit files like ltsp.conf or ltsp-dnsmasq.conf etc.

So (2) isn't really important, and I'm not sure it would be a valuable application if it wasn't for (1).
I would love to see (1) become a separate Debian package; I think @enaut had started maintaining LTSP manager for that, not sure what the current status is.

**Comment by rkwesk** [comment]
I agree with everything you've written except for:

"So (2) isn't really important, and I'm not sure it would be a valuable application if it wasn't for (1)."

For those who have not yet used (or even heard of) this project (ltsp) these menu options are the only (practically speaking) way they can quickly activate whatever commands are necessary at any given time. I would go further and recommend that rather than simply having a button for each command separately that there be mini scripts behind some so that groups of commands would be run in the appropriate order. Two examples jump to mind while I’m writing this: one is that ltsp image be followed by ltsp initrd, two is that behind the buttons to edit the confs that ltsp initrd follow any such edits.

Almost everyone I have worked with has little time and practically no understanding behind how all this works or how to cope with it.

Richard

**Comment by enaut** [comment]
I maintain the ltsp-manager in my personal ppa... I haven't found the time to test it with the new ltsp yet but it should just work (appart from the configurationfile menu entries). But you should be able to batchcreate users and delete them assign groups change passwords etc.

As for the future I'm not really sure how to handle the package. Currently this is a script run by root and doing everything as priviledged user. I'd like to make it user based and use dbus with a priviledged service...

Also I think it should be capable of doing all its stuff via ssh. So that the teacher computer does not have to be the same as the server. For me the server is a headless system in the basement.
The things why I stopped implementing the above are:
  * https://gitlab.freedesktop.org/accountsservice/accountsservice/issues/64
  * very limited time on my part
  * ppa and deb is very complicated - just why? :(

**Comment by alkisg** [comment]
Hmm I think the goal of doing things via ssh (=remotely), conflicts with the goal of doing things via dbus (=locally).

With the current design of just running usermod commands instead of using accountsservice, one could easily run the GUI as a user, and add a "proxycommand" option.

- If proxycommand="sudo", then the commands would be ran locally.
- If proxycommand="ssh server sudo", then the commands would be ran remotely.

In both cases the user would need to be a sudoer though (or use setuid; and in the second case, also have passwordless ssh set up).

Accountsservice was supposed to be replaced by sssd at some point in the future, not sure what happened with that plan.

Another way of doing things remotely is a web based interface; I think some tools already exist for that; the initial idea of ltsp-manager was to be a gtk application instead of html/javascript though.
